### PR TITLE
[diffusion] doc: update Cosmos3 cookbook

### PR DESCRIPTION
**Pull request by rwang5203**
## Summary

- Add a Cosmos3 guide for T2I, T2V, I2V, V2V, sound, and action modes.
- Document the FlowUniPC defaults and offline generation parameters.
- Register the guide in the current `docs_new` navigation.

The runtime implementation is already merged in #27168. After rebasing onto that change, this PR is documentation-only.

## Testing

- `pre-commit run --from-ref origin/main --to-ref HEAD`

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #29066263796](https://github.com/sgl-project/sglang/actions/runs/29066263796)
Latest PR Test (Extra): :white_check_mark: [Run #29066263740](https://github.com/sgl-project/sglang/actions/runs/29066263740)

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/models/dits/action_utils.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    """Canonical ``"W,H"`` aspect string, matching the reference table lookup."""
+    for sizes in VIDEO_RES_SIZE_INFO.values():
+        for aspect, (cand_w, cand_h) in sizes.items():
+            if width == cand_w and height == cand_h:
+                return aspect
+    divisor = math.gcd(width, height)
+    return f"{width // divisor},{height // divisor}"
+
+
+def build_action_prompt(
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/models/dits/action_utils.py]: is this only used by test?

File: python/sglang/multimodal_gen/runtime/layers/domain_aware_linear.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: Apache-2.0
+"""Per-embodiment-domain linear projection for Cosmos3 action heads."""
+
+import torch
+from torch import nn
+
+
+class DomainAwareLinear(nn.Module):
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/layers/domain_aware_linear.py]: please put nn.Module layers under `layers/`

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py [hunk 3]
```diff
@@ -350,6 +503,148 @@ def forward(self, batch: Req, server_args: ServerArgs) -> Req:
...
+            raise ValueError(
+                f"Unknown Cosmos3 action_mode {mode!r}. Expected one of "
+                f"{list(self.VALID_MODES)}."
+            )
+
+        device = get_local_torch_device()
+        dtype = torch.bfloat16
+        embodiment = getattr(batch, "action_embodiment", "droid_lerobot")
+        domain_id = get_domain_id(embodiment)
+        raw_dim = get_raw_action_dim(embodiment)
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py]: if methods like `get_raw_action_dim` are only used here, I'd prefer inlining them in this file for now

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py [hunk 4]
```diff
@@ -350,6 +503,148 @@ def forward(self, batch: Req, server_args: ServerArgs) -> Req:
...
+            generator = torch.Generator(device=device).manual_seed(batch.seed)
+        shape = (1, self.transformer.sound_dim, t_sound)
+        batch.extra["sound_latents"] = torch.randn(
+            shape, generator=generator, device=device, dtype=torch.bfloat16
+        )
+        self.log_info(f"Prepared sound latents with shape {shape}")
+        return batch
+
+
+class Cosmos3ActionLatentPreparationStage(PipelineStage):
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py]: could we try to inherit from `LatentPreparationStage`?

File: python/sglang/multimodal_gen/runtime/models/dits/action_utils.py (hunk 1 above)
Comment (rwang5203) [reply python/sglang/multimodal_gen/runtime/models/dits/action_utils.py]: No. `build_action_prompt` uses it for the caption `aspect_ratio` field, and the test covers it too. Every `action_utils` symbol was used only by this stage, so it now lives in `cosmos3.py` after I folded the module in there.

File: python/sglang/multimodal_gen/runtime/layers/domain_aware_linear.py (hunk 2 above)
Comment (rwang5203) [reply python/sglang/multimodal_gen/runtime/layers/domain_aware_linear.py]: Done. Moved to `runtime/layers/domain_aware_linear.py` and updated the import.

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py (hunk 3 above)
Comment (rwang5203) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py]: Done. `action_utils` was used only by this stage, so I folded it into `cosmos3.py` and deleted the module. I kept `get_domain_id` / `get_raw_action_dim` / `load_action_stats` as small named helpers rather than inlining at the call sites, since each validates its arg and raises a typed error (and `load_action_stats` is reused). Happy to inline the lookups if you prefer.

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py (hunk 4 above)
Comment (rwang5203) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py]: Tried it, poor fit. That base targets the main video track and needs a scheduler plus grouped-request, latent-spec, noise-scaling, and length-adjust logic that a 1D sound latent or fixed action chunk does not use. Even the main `Cosmos3LatentPreparationStage` stays on `PipelineStage`, so I kept these aux stages there too. Open to a shared light base if you prefer.
